src/ops.rs

```diff
@@ -201,3 +201,48 @@ impl_op_unary! { impl Neg, neg for  Matrix<T, M, N> }
 impl_op_unary! { impl Neg, neg for &Matrix<T, M, N> }
 impl_op_unary! { impl Not, not for  Matrix<T, M, N> }
 impl_op_unary! { impl Not, not for &Matrix<T, M, N> }
+
+////////////////////////////////////////////////////////////////////////////////
+// Matrix * Matrix
+////////////////////////////////////////////////////////////////////////////////
+
+impl<T, const N: usize, const M: usize> Matrix<T, N, M> {
+    pub(crate) fn matmul_into<const P: usize>(
+        &self,
+        rhs: &Matrix<T, M, P>,
+        out: &mut Matrix<T, N, P>,
```

Comment:
I think the mutable reference is not necessary, this function can construct it using `Matrix::default()` since the `Output` for each implementation is the same thing.

Reply:
The idea behind this, and something I've tested in a crate of mine is to (unsafely) cast slices to `&mut Matrix` and compute successive matrix multiplications directly into the buffers